nixos/modules/tasks/network-namespaces.nix

```diff
@@ -0,0 +1,44 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.networking;
+in
+{
+  options = {
+    networking.namespaces = mkOption {
+      default = [];
+      type = types.listOf types.str;
+      description = ''
+        A list of named network namespaces to create.
+      '';
+    };
+  };
+  config = mkIf (builtins.length cfg.namespaces != 0) {
+    systemd.services.network-namespaces = {
+      description = "Named network namespaces";
+      before = [ "network-pre.target" ];
+      wantedBy = [ "multi-user.target" ];
+      path = [ pkgs.iproute ];
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = true;
+        ExecStart = pkgs.writeScript "netns-up" ''
+          #! ${pkgs.bash}/bin/bash
+          for netns in ${toString cfg.namespaces}
+          do
+            ip netns add $netns
```

Comment:
please escape or at least quote the netns name here. We don't want funny things to happen. Same applies for the teardown.

Reply:
done